interfaces/apparmor/backend.go

```diff
@@ -216,7 +217,11 @@ func (b *Backend) deriveContent(spec *Specification, snapInfo *snap.Info, opts i
 
 func addContent(securityTag string, snapInfo *snap.Info, opts interfaces.ConfinementOptions, snippetForTag string, content map[string]*osutil.FileState) {
 	var policy string
-	if opts.Classic && !opts.JailMode {
+	// When apparmor partial then use the classic template as we cannot enable
+	// effective confinement and we cannot rely on the devmode template which
+	// may have things that local apparmor doesn't understand.
+	level, _ := aa.ProbeKernel().Evaluate()
```

Comment:
Why are we using aa.ProbeKernel().Evaluate()" instead of just "aa.Evaluate()" and hidding the "ProbeKernel()" as an implementation detail in `aa` ?

Reply:
Because later on we'll get the result of ProbeKernel from the backend itself and specific interfaces will be able to get it from there.